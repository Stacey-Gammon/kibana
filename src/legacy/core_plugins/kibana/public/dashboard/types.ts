/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import { AppState } from 'ui/state_management/app_state';
import { Query, Filter, ViewMode } from 'plugins/embeddable_api';

export interface GridData {
  w: number;
  h: number;
  x: number;
  y: number;
  i: string;
}

export interface SavedDashboardPanel {
  readonly id?: string;
  readonly version: string;
  readonly type: string;
  readonly panelIndex: string;
  readonly embeddableConfig: any;
  readonly gridData: GridData;
  readonly title?: string;
}

export interface Pre61SavedDashboardPanel {
  readonly size_x: number;
  readonly size_y: number;
  readonly row: number;
  readonly col: number;
  readonly panelIndex: any; // earlier versions allowed this to be number or string
  readonly id: string;
  readonly type: string;
}

export interface Pre64SavedDashboardPanel {
  columns?: string;
  sort?: string;
  readonly id?: string;
  readonly version: string;
  readonly type: string;
  readonly panelIndex: string;
  readonly gridData: GridData;
  readonly title?: string;
  readonly embeddableConfig: any;
}

export interface DashboardAppStateDefaults {
  panels: SavedDashboardPanel[];
  fullScreenMode: boolean;
  title: string;
  description?: string;
  timeRestore: boolean;
  options: {
    useMargins: boolean;
    hidePanelTitles: boolean;
  };
  query: Query;
  filters: Filter[];
  viewMode: ViewMode;
}

export interface DashboardAppState extends AppState {
  panels: SavedDashboardPanel[];
  fullScreenMode: boolean;
  title: string;
  description: string;
  timeRestore: boolean;
  options: {
    hidePanelTitles: boolean;
    useMargins: boolean;
  };
  query: Query;
  filters: Filter[];
  viewMode: ViewMode;
}
