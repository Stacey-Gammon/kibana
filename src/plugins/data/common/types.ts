/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import { SearchParams } from 'elasticsearch';

export * from './timefilter/types';

export interface SearchArguments {
  // The Elasticsearch search parameters
  searchParams: SearchParams;

  // An optional signal to abort the request
  signal?: AbortSignal;

  // Additional search options to customize the behavior of the request
  options?: SearchOptions;
}

export interface SearchOptions {
  [propName: string]: any;

  // If courier:setRequestPreference is set to "sessionId", this will be used as the preference
  sessionId?: string;
}
