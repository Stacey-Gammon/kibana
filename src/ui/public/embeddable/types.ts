/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

export interface ContainerState {
  // 'view' or 'edit'. Should probably be an enum but I'm undecided where to define it, here or in dashboard code.
  viewMode: string;

  timeRange: {
    // To and From should be either an absolute time range in utc format or a relative one (e.g. now-15m)
    to: string;
    from: string;
  };

  // The shape will be up to the embeddable type.
  embeddableCustomization?: object;

  hidePanelTitles: boolean;

  /**
   * Is the current panel in expanded mode
   */
  isPanelExpanded: boolean;
}

export interface EmbeddableState {
  /**
   * Any customization data that should be stored at the panel level. For
   * example, pie slice colors, or custom per panel sort order or columns.
   */
  customization: object;
  /**
   * A possible filter the embeddable wishes dashboard to apply.
   */
  stagedFilter: object;
}
