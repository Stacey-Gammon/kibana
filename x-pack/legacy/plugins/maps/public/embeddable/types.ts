/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */
export interface MapFeature {
  id: number;
  layerId: string;
}

export interface LoadFeatureProps {
  layerId: string;
  featureId: number;
}

export interface FeatureProperty {
  _propertyKey: string;
  _rawValue: string | string[];
}

export interface FeatureGeometry {
  coordinates: [number];
  type: string;
}

export interface RenderTooltipContentParams {
  addFilters(filter: object): void;
  closeTooltip(): void;
  features: MapFeature[];
  isLocked: boolean;
  getLayerName(layerId: string): Promise<string>;
  loadFeatureProperties({ layerId, featureId }: LoadFeatureProps): Promise<FeatureProperty[]>;
  loadFeatureGeometry({ layerId, featureId }: LoadFeatureProps): FeatureGeometry;
}
