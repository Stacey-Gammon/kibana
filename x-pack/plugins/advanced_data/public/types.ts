/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { ISyncSearchRequest, ISearchOptions } from 'src/plugins/data/public';
import { IKibanaSearchRequest, IKibanaSearchResponse } from '../../../../src/plugins/data/common';

export interface ISqlSearchRequest extends IKibanaSearchRequest {
  sql: string;
}

export interface ISqlSearchResponse extends IKibanaSearchResponse {
  results: any;
}

export interface IDemoDataRequest extends IKibanaSearchRequest {
  responseTime: number;
  totalHitCount: number;
}

export interface IDemoDataHit {
  title: string;
  message: string;
}

export interface IDemoDataResponse extends IKibanaSearchResponse {
  hits: IDemoDataHit[];
}

export interface IAsyncSearchRequest extends ISyncSearchRequest {
  /**
   * The ID received from the response from the initial request
   */
  id?: string;
}

export interface IAsyncSearchOptions extends ISearchOptions {
  /**
   * The number of milliseconds to wait between receiving a response and sending another request
   */
  pollInterval?: number;
}
