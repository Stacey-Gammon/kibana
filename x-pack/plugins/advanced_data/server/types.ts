/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { IKibanaSearchRequest, IKibanaSearchResponse } from 'src/plugins/data/common';
import { ISearchStrategy } from '../../../../src/plugins/data/server';
import { ASYNC_DEMO_SEARCH_STRATEGY } from '../common';

export interface ISqlSearchRequest extends IKibanaSearchRequest {
  sql: string;
}

export interface ISqlSearchResponse extends IKibanaSearchResponse {
  results: any;
}

export interface IDemoDataRequest extends IKibanaSearchRequest {
  responseTime: number;
  totalHitCount: number;
}

export interface IDemoDataHit {
  title: string;
  message: string;
}

export interface IDemoDataResponse extends IKibanaSearchResponse {
  hits: IDemoDataHit[];
  responseTime?: number;
}

export interface ISearchesInProgress {
  [id: string]: {
    request: IDemoDataRequest;
    response: IDemoDataResponse;
    requestStartTime: number;
    total: number;
    loaded: number;
    responseTime: number;
  };
}

export type TDemoDataSearchStrategyProvider = (
  searchesInProgress: ISearchesInProgress
) => ISearchStrategy<typeof ASYNC_DEMO_SEARCH_STRATEGY>;
