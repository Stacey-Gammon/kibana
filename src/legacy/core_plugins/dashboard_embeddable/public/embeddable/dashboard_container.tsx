/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import ReactDOM from 'react-dom';
import uuid from 'uuid';

import { I18nProvider } from '@kbn/i18n/react';
import { IndexPattern } from 'ui/index_patterns';

import {
  Container,
  ContainerInput,
  ContainerOutput,
  Embeddable,
  EmbeddableFactoryRegistry,
  EmbeddableInput,
  Filter,
  Query,
  RefreshConfig,
  ViewMode,
  EmbeddableOutput,
} from 'plugins/embeddable_api/index';
import { TimeRange } from 'ui/timefilter/time_history';

import { EmbeddableInputMissingFromContainer } from 'plugins/embeddable_api/containers/container';
import { DASHBOARD_CONTAINER_TYPE } from './dashboard_container_factory';
import { createPanelState } from './panel';
import { DashboardPanelState } from './types';
import { DashboardViewport } from './viewport/dashboard_viewport';

export interface DashboardContainerInput extends ContainerInput {
  viewMode: ViewMode;
  filters: Filter[];
  query: Query;
  timeRange: TimeRange;
  refreshConfig?: RefreshConfig;
  expandedPanelId?: string;
  useMargins: boolean;
  title: string;
  description?: string;
  isFullScreenMode: boolean;
  panels: { [panelId: string]: DashboardPanelState };

  // Used to force a refresh of embeddables even if there were no other input state
  // changes.
  lastReloadRequestTime?: number;
}

export interface DashboardEmbeddableInput extends EmbeddableInput {
  // customization: any;
  filters: Filter[];
  isPanelExpanded: boolean;
  query: Query;
  timeRange: TimeRange;
  refreshConfig?: RefreshConfig;
  viewMode: ViewMode;
  savedObjectId?: string;
  hidePanelTitles?: boolean;
}

export interface DashboardEmbeddableOutput extends EmbeddableOutput {
  indexPatterns?: IndexPattern[];
}

export type DashboardEmbeddable = Embeddable<DashboardEmbeddableInput, DashboardEmbeddableOutput>;

export class DashboardContainer extends Container<
  DashboardEmbeddableInput,
  DashboardEmbeddableOutput,
  DashboardContainerInput
> {
  constructor(
    initialInput: DashboardContainerInput,
    embeddableFactories: EmbeddableFactoryRegistry
  ) {
    super(DASHBOARD_CONTAINER_TYPE, initialInput, { embeddableLoaded: {} }, embeddableFactories);
  }

  public createNewPanelState<EEI extends EmbeddableInput = EmbeddableInput>({
    type,
    initialInput,
  }: {
    type: string;
    initialInput: EmbeddableInputMissingFromContainer<EEI, DashboardEmbeddableInput> & {
      id?: string;
    };
  }): DashboardPanelState<EmbeddableInputMissingFromContainer<EEI, DashboardEmbeddableInput>> {
    const embeddableId = initialInput.id || uuid.v4();
    return createPanelState<EmbeddableInputMissingFromContainer<EEI, DashboardEmbeddableInput>>(
      { ...initialInput, id: embeddableId },
      type,
      Object.values(this.input.panels)
    );
  }

  public onToggleExpandPanel(id: string) {
    const newValue = this.input.expandedPanelId ? undefined : id;
    this.updateInput({
      expandedPanelId: newValue,
    });
  }

  public onPanelsUpdated = (panels: { [panelId: string]: DashboardPanelState }) => {
    this.updateInput({
      panels: {
        ...panels,
      },
    });
  };

  public onExitFullScreenMode = () => {
    this.updateInput({
      isFullScreenMode: false,
    });
  };

  public render(dom: React.ReactNode) {
    ReactDOM.render(
      // @ts-ignore
      <I18nProvider>
        <DashboardViewport embeddableFactories={this.embeddableFactories} container={this} />
      </I18nProvider>,
      dom
    );
  }

  public getPanelIndexPatterns() {
    const indexPatterns: IndexPattern[] = [];
    Object.values(this.embeddables).forEach(embeddable => {
      const embeddableIndexPatterns = embeddable.getOutput().indexPatterns;
      if (embeddableIndexPatterns) {
        indexPatterns.push(...embeddableIndexPatterns);
      }
    });
    return indexPatterns;
  }

  public getInputForEmbeddableFromContainer(id: string): DashboardEmbeddableInput {
    const panelState = this.getPanelState(id);
    const isPanelExpanded = this.input.expandedPanelId === panelState.embeddableId;
    const { viewMode, refreshConfig, timeRange, query, hidePanelTitles, filters } = this.input;
    return {
      filters,
      hidePanelTitles,
      isPanelExpanded,
      query,
      timeRange,
      refreshConfig,
      viewMode,
      id: panelState.embeddableId,
      customization: panelState.customization,
    };
  }
}
