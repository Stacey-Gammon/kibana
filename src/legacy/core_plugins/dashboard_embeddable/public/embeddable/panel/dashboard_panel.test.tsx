/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

jest.mock('ui/metadata', () => ({
  metadata: {
    branch: 'my-metadata-branch',
    version: 'my-metadata-version',
  },
}));

import {
  GreetingEmbeddable,
  GreetingEmbeddableInput,
  GreetingEmbeddableOutput,
} from 'plugins/embeddable_api/__test__';
import { embeddableFactories } from 'plugins/embeddable_api/index';
import { QueryLanguageType, ViewMode } from 'plugins/embeddable_api/types';
import React from 'react';
import { mountWithIntl, nextTick } from 'test_utils/enzyme_helpers';
import { DashboardContainer, DashboardContainerInput } from '../dashboard_container';
import { DashboardPanel } from './dashboard_panel';
import { GREETING_EMBEDDABLE } from 'plugins/embeddable_api/__test__';

function getDashboardContainerInput(): DashboardContainerInput {
  return {
    id: '123',
    viewMode: ViewMode.EDIT,
    filters: [],
    query: {
      language: QueryLanguageType.KUERY,
      query: '',
    },
    timeRange: {
      to: 'now',
      from: 'now-15m',
    },
    useMargins: false,
    title: 'My Test Dashboard',
    isFullScreenMode: false,
    panels: {},
  };
}

test('DashboardPanel renders an embeddable when it is done loading', async () => {
  const container = new DashboardContainer(getDashboardContainerInput(), embeddableFactories);
  const newEmbeddable = await container.addNewEmbeddable<
    GreetingEmbeddableInput,
    GreetingEmbeddableOutput,
    GreetingEmbeddable
  >(GREETING_EMBEDDABLE, {
    firstName: 'Theon',
    lastName: 'Greyjoy',
    id: '123',
  });

  expect(newEmbeddable.id).toBeDefined();

  const component = mountWithIntl(
    <DashboardPanel.WrappedComponent
      intl={null as any}
      container={container}
      embeddableId={newEmbeddable.id}
    />
  );

  await nextTick();
  component.update();

  // Due to the way embeddables mount themselves on the dom node, they are not forced to be
  // react components, and hence, we can't use the usual
  // findTestSubject(component, 'dashboardPanelHeading-HelloTheonGreyjoy');
  expect(
    component
      .getDOMNode()
      .querySelectorAll('[data-test-subj="dashboardPanelHeading-HelloTheonGreyjoy"]').length
  ).toBe(1);
});
