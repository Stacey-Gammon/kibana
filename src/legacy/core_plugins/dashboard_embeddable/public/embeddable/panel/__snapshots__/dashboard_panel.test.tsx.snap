// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardPanel renders an embeddable when it is done loading 1`] = `
<DashboardPanelUi
  container={
    DashboardContainer {
      "anyChangeListeners": Array [],
      "container": undefined,
      "embeddableFactories": EmbeddableFactoryRegistry {
        "factories": Object {
          "dashboard": DashboardContainerFactory {
            "name": "dashboard",
            "savedObjectMetaData": undefined,
          },
          "hello_world": HelloWorldEmbeddableFactory {
            "name": "hello_world",
            "savedObjectMetaData": undefined,
          },
        },
      },
      "embeddableUnsubscribes": Object {
        "739ca72c-f593-4d87-8a06-1daa12620bf9": [Function],
      },
      "embeddables": Object {
        "739ca72c-f593-4d87-8a06-1daa12620bf9": HelloWorldEmbeddable {
          "anyChangeListeners": Array [],
          "container": [Circular],
          "id": "739ca72c-f593-4d87-8a06-1daa12620bf9",
          "input": Object {},
          "inputChangeListeners": Array [],
          "isContainer": false,
          "output": Object {
            "title": "Hello World!",
          },
          "outputChangeListeners": Array [
            [Function],
          ],
          "panelContainer": <div
            class="embPanel panel-content"
            id="embeddedPanel"
          >
            <div
              class="euiPanel embPanel embPanel--editing"
              data-test-subj="embeddablePanel"
            >
              <div
                class="embPanel__header"
                data-test-subj="dashboardPanelHeading-HelloWorld!"
              >
                <div
                  aria-label="Dashboard panel: Hello World!"
                  class="embPanel__title embPanel__dragger"
                  data-test-subj="dashboardPanelTitle"
                  title="Hello World!"
                >
                  Hello World!
                </div>
                <div
                  class="euiPopover euiPopover--anchorDownRight euiPopover--withTitle embPanel_optionsMenuPopover"
                  data-test-subj="dashboardPanelContextMenuClosed"
                  id="dashboardPanelContextMenu"
                >
                  <div
                    class="euiPopover__anchor"
                  >
                    <button
                      aria-label="Panel options"
                      class="euiButtonIcon euiButtonIcon--text embPanel_optionsMenuButton"
                      data-test-subj="dashboardPanelToggleMenuIcon"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="euiIcon euiIcon--medium euiButtonIcon__icon"
                        focusable="false"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                        xmlns:xlink="http://www.w3.org/1999/xlink"
                      >
                        <defs>
                          <path
                            d="M.164 10.329L1.87 8 .163 5.67c.18-.601.43-1.19.758-1.757a8.197 8.197 0 0 1 1.142-1.535l2.872.313L6.099.05a8.166 8.166 0 0 1 3.8-.003l1.166 2.644 2.872-.313a8.166 8.166 0 0 1 1.899 3.293L14.13 8l1.706 2.33c-.18.601-.43 1.19-.758 1.757a8.197 8.197 0 0 1-1.142 1.535l-2.872-.313-1.164 2.641a8.166 8.166 0 0 1-3.8.003l-1.166-2.644-2.872.313a8.166 8.166 0 0 1-1.899-3.293zm4.663 1.986a1 1 0 0 1 1.023.591l.957 2.17c.79.134 1.597.132 2.387-.001l.956-2.169a1 1 0 0 1 1.023-.59l2.358.256a7.23 7.23 0 0 0 1.194-2.068l-1.401-1.913a1 1 0 0 1 0-1.182l1.4-1.912a7.165 7.165 0 0 0-1.192-2.069l-2.359.257a1 1 0 0 1-1.023-.591L9.193.924a7.165 7.165 0 0 0-2.387.001L5.85 3.094a1 1 0 0 1-1.023.59l-2.358-.256a7.23 7.23 0 0 0-1.194 2.068l1.401 1.913a1 1 0 0 1 0 1.182l-1.4 1.912c.28.751.681 1.45 1.192 2.069l2.359-.257zM8 11a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm0-1a2 2 0 1 0 0-4 2 2 0 0 0 0 4z"
                            id="gear-a"
                          />
                        </defs>
                        <use
                          xlink:href="#gear-a"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="embeddable-root panel-content"
              >
                <div>
                  Hello World!
                </div>
              </div>
            </div>
          </div>,
          "type": "hello_world",
        },
      },
      "id": "f38730d4-e122-4544-b68b-9d8cf6316dd5",
      "input": Object {
        "filters": Array [],
        "isFullScreenMode": false,
        "panels": Object {
          "739ca72c-f593-4d87-8a06-1daa12620bf9": Object {
            "embeddableId": "739ca72c-f593-4d87-8a06-1daa12620bf9",
            "gridData": Object {
              "h": 15,
              "i": "739ca72c-f593-4d87-8a06-1daa12620bf9",
              "w": 24,
              "x": 0,
              "y": 0,
            },
            "initialInput": Object {},
            "type": "hello_world",
          },
        },
        "query": Object {
          "language": "kuery",
          "query": "",
        },
        "timeRange": Object {
          "from": "now-15m",
          "to": "now",
        },
        "title": "My Test Dashboard",
        "useMargins": false,
        "viewMode": "edit",
      },
      "inputChangeListeners": Array [],
      "isContainer": true,
      "onExitFullScreenMode": [Function],
      "onPanelsUpdated": [Function],
      "output": Object {
        "embeddableLoaded": Object {
          "739ca72c-f593-4d87-8a06-1daa12620bf9": true,
        },
        "filters": Array [],
        "isFullScreenMode": false,
        "panels": Object {},
        "query": Object {
          "language": "kuery",
          "query": "",
        },
        "timeRange": Object {
          "from": "now-15m",
          "to": "now",
        },
        "title": "My Test Dashboard",
        "useMargins": false,
        "viewMode": "edit",
      },
      "outputChangeListeners": Array [
        [Function],
        [Function],
      ],
      "panelContainer": undefined,
      "type": "dashboard",
    }
  }
  embeddableId="739ca72c-f593-4d87-8a06-1daa12620bf9"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {
        "date": Object {
          "full": Object {
            "day": "numeric",
            "month": "long",
            "weekday": "long",
            "year": "numeric",
          },
          "long": Object {
            "day": "numeric",
            "month": "long",
            "year": "numeric",
          },
          "medium": Object {
            "day": "numeric",
            "month": "short",
            "year": "numeric",
          },
          "short": Object {
            "day": "numeric",
            "month": "numeric",
            "year": "2-digit",
          },
        },
        "number": Object {
          "currency": Object {
            "style": "currency",
          },
          "percent": Object {
            "style": "percent",
          },
        },
        "relative": Object {
          "days": Object {
            "units": "day",
          },
          "hours": Object {
            "units": "hour",
          },
          "minutes": Object {
            "units": "minute",
          },
          "months": Object {
            "units": "month",
          },
          "seconds": Object {
            "units": "second",
          },
          "years": Object {
            "units": "year",
          },
        },
        "time": Object {
          "full": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "long": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "medium": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
          },
          "short": Object {
            "hour": "numeric",
            "minute": "numeric",
          },
        },
      },
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": null,
    }
  }
>
  <div
    className="embPanel panel-content"
    id="embeddedPanel"
  />
</DashboardPanelUi>
`;
